File: v3-sandbox/prototype-java/src/test/java/org/hiero/common/AccountIdTest.java
package org.hiero.common;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

class AccountIdTest {

    @Test
    void parseShardRealmNum() {
        // given
        final String input = "0.0.1234";

        // when
        final AccountId id = AccountId.fromString(input);

        // then
        assertEquals(0L, id.shard());
        assertEquals(0L, id.realm());
        assertEquals(1234L, id.num());
        assertEquals("", id.checksum());
        assertEquals("0.0.1234", id.toString());
    }

    @Test
    void parseWithChecksum() {
        // given
        final String input = "0.0.1234-abcde";

        // when
        final AccountId id = AccountId.fromString(input);

        // then
        assertEquals("abcde", id.checksum());
        assertEquals("0.0.1234-abcde", id.toStringWithChecksum());
    }

    @Test
    void rejectsNegative() {
        assertThrows(IllegalArgumentException.class, () -> new AccountId(-1L, 0L, 0L));
    }

    @Test
    void rejectsInvalidString() {
        assertThrows(IllegalArgumentException.class, () -> AccountId.fromString("not-an-id"));
    }
}

File: v3-sandbox/prototype-typescript/test/common/AccountId.test.ts
import { describe, expect, it } from "vitest";
import { AccountId } from "../../src/common/index.js";

describe("AccountId", () => {
  it("parses shard.realm.num format", () => {
    const id = AccountId.fromString("0.0.123456");
    expect(id.shard).toBe(0n);
    expect(id.realm).toBe(0n);
    expect(id.num).toBe(123456n);
    expect(id.checksum).toBe("");
  });

  it("parses with checksum suffix", () => {
    const id = AccountId.fromString("0.0.123456-xxxxx");
    expect(id.num).toBe(123456n);
    expect(id.checksum).toBe("xxxxx");
  });

  it("toString returns shard.realm.num", () => {
    expect(AccountId.fromString("0.0.5").toString()).toBe("0.0.5");
  });

  it("toStringWithChecksum includes checksum when present", () => {
    const id = AccountId.fromString("0.0.5-abcde");
    expect(id.toStringWithChecksum()).toBe("0.0.5-abcde");
  });

  it("toStringWithChecksum omits dash when no checksum", () => {
    expect(AccountId.fromString("0.0.5").toStringWithChecksum()).toBe("0.0.5");
  });

  it("throws on invalid format", () => {
    expect(() => AccountId.fromString("not-an-id")).toThrow("illegal-format");
    expect(() => AccountId.fromString("0.0")).toThrow("illegal-format");
  });
});
